rust/tools/authorization-logic/src/souffle/datalog_ir.rs

```diff
@@ -18,14 +18,20 @@
 //! this authorization logic into datalog simpler.
 use crate::ast::*;
 
+#[derive(Clone)]
+pub enum DLIRRValue {
+    PredicateRValue { predicate: AstPredicate },
+    BinopRValue { binop: AstBinaryOperation }
```

Comment:
This comment is mostly about naming. It looks `DLIRRValue` is actually a predicate and you have two kinds of predicates:
   - Datalog Relations.
   - Arithmetic comparisons. 

Why can't you call it `DLIRPredicate` or something like that?

Reply:
I think we need some name other than predicate to capture the whole syntax of objects that can appear on the RHS. Predicate is just one of those kinds of values and the arithmetic operations are another. Maybe in the future we extend this syntax and there are even more kinds of objects that can appear on the RHS of a conditional. I thought `RValue` was an intuitive way to name this part of the grammar for the reasons described in the other thread.

Reply:
Sounds good. Don't want to push back too much on this, but I understand what these are. Can you add some brief documentation?

Reply:
fixed